src/clients/python/experimental_api_v2/library/grpcclient.py

```diff
@@ -0,0 +1,273 @@
+        self._client_stub = grpc_service_v2_pb2_grpc.GRPCInferenceServiceStub(
+            self._channel)
+        self._verbose = verbose
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.close()
+
+    def __del__(self):
+        self.close()
+
+    def close(self):
+        """Close the client. Any future calls to server
+        will result in an Error.
+
+        """
+        self._channel.close()
+
+    def is_server_live(self):
+        """Contact the inference server and get liveness.
+
+        Returns
+        -------
+        bool
+            True if server is live, False if server is not live.
+
+        Raises
+        ------
+        InferenceServerException
+            If unable to get liveness.
+
+        """
+        try:
+            self._request = grpc_service_v2_pb2.ServerLiveRequest()
+            self._response = self._client_stub.ServerLive(self._request)
+            return self._response.live
+        except grpc.RpcError as rpc_error:
+            raise_error_grpc(rpc_error)
+
+    def is_server_ready(self):
+        """Contact the inference server and get readiness.
+
+        Returns
+        -------
+        bool
+            True if server is ready, False if server is not ready.
+
+        Raises
+        ------
+        InferenceServerException
+            If unable to get readiness.
+
+        """
+        try:
+            self._request = grpc_service_v2_pb2.ServerReadyRequest()
+            self._response = self._client_stub.ServerReady(self._request)
+            return self._response.ready
+        except grpc.RpcError as rpc_error:
+            raise_error_grpc(rpc_error)
+
+    def is_model_ready(self, model_name, model_version=-1):
+        """Contact the inference server and get the readiness of specified model.
+
+         Parameters
+        ----------
+        model_name: str
+            The name of the model to check for readiness.
+
+        model_version: int
+            The version of the model to check for readiness. If -1 is given the 
+            server will choose a version based on the model and internal policy.
+
+
+        Returns
+        -------
+        bool
+            True if the model is ready, false if not ready.
+
+        Raises
+        ------
+        InferenceServerException
+            If unable to get model readiness.
+
+        """
+        try:
+            self._request = grpc_service_v2_pb2.ModelReadyRequest(
+                name=model_name, version=model_version)
+            self._response = self._client_stub.ModelReady(self._request)
+            return self._response.ready
+        except grpc.RpcError as rpc_error:
+            raise_error_grpc(rpc_error)
+
+    def get_server_metadata(self):
+        """Contact the inference server and get its metadata.
+
+        Returns
+        -------
+        dict
+            The JSON object holding the metadata
```

Comment:
Since this is the GRPC client API there should be a way to get the protobuf objects directly. Perhaps have an option to request JSON return value that defaults to false.

Same for other methods below.